language/move-ir/types/src/location.rs

```diff
@@ -2,29 +2,43 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use codespan::Span;
+use once_cell::sync::Lazy;
 use serde::{Deserialize, Serialize};
 use std::{
     cmp::Ordering,
     fmt,
     hash::{Hash, Hasher},
+    sync::Mutex,
 };
 
 //**************************************************************************************************
 // Loc
 //**************************************************************************************************
 
+pub static FILENAME_CACHE: Lazy<Mutex<Vec<String>>> = Lazy::new(|| Mutex::new(vec![]));
```

Comment:
Probably want this in another Rust module

Reply:
Like we discussed in chat, I'll probably stack these commits such that the symbol table is added first. But, for now as part of this WIP, I added a commit on top of the stack that adds a new Rust module, `symbol-table`.

Reply:
Needs to be removed after the switch :)